packages/dev/core/src/Events/pointerEvents.ts

```diff
@@ -109,21 +110,34 @@ export class PointerInfoPre extends PointerInfoBase {
  * The event member is an instance of PointerEvent for all types except PointerWheel and is of type MouseWheelEvent when type equals PointerWheel. The different event types can be found in the PointerEventTypes class.
  */
 export class PointerInfo extends PointerInfoBase {
+    private _pickInfo: Nullable<PickingInfo>;
+    private _inputManager: Nullable<InputManager>;
+
+    /**
+     * Defines the picking info associated to the info (if any)
+     */
+    public get pickInfo(): Nullable<PickingInfo> {
+        if (this._inputManager) {
+            // Store InputManager locally so that we can null out the object ref and avoid circular reference
+            const inputManager = this._inputManager;
+            this._inputManager = null;
+            this._pickInfo = inputManager._pickMove((this.event as IPointerEvent).pointerId);
+            inputManager._setRayOnPointerInfo(this);

```

Comment:
Consider changing `_setRayOnPointerInfo` to take `pickInfo` and `event` instead of `pointerInfo` to avoid circular reference.